### PR TITLE
Added doc for modification of action delete API

### PR DESCRIPTION
**Pull request by mahesh-orch**
Adding documentation for modification of action delete API to remove related files on disk and added arguments `-f`, `--force` as auto delete flags to action delete command on CLI to remove disk files without asking user for permission.

### CODE REVIEW

**Comment by blag** [comment]
@mahesh-orch Would you please use more descriptive commit messages, and minimize the number of commits you generate? For every single commit that you push, many people get an email from GitHub including the commit message, and a bunch of commits becomes a deluge of email notifications.

**Comment by mahesh-orch** [comment]
@blag Sure, I will keep commits minimum and descriptive. Thanks.

**Comment by amanda11** [comment]
@mahesh-orch For the documentation, we will also need to make sure we add something to the upgrade_notes.rst. To open up a new section for 3.6, so that we make it clear that there are backwards incompatible changes on deleting action through the CLI and API.

**Comment by mahesh-orch** [comment]
@amanda11 I have added new section for st2 v3.6 and added a note for made change in action delete. Please review. Thanks.

File: docs/source/upgrade_notes.rst
Comment (amanda11) [inline docs/source/upgrade_notes.rst:12]: @mahesh-orch I'd maybe rephrase slightly on the CLI command. I'd mention that the action delete CLI will now wait for approval to delete on disk before proceeding, unless you use one of the -f or --force commands. As we want to get across the difference to those already using that command. 
If they are using it in a script they need to know that they would need to update to use the one of the new parameters, otherwise their existing scripts would need updating.

File: docs/source/actions.rst
Comment (amanda11) [inline docs/source/actions.rst:77]: Suggest replace "Previously" with "Prior to 3.6 release,"

File: docs/source/actions.rst
Comment (amanda11) [inline docs/source/actions.rst:78]: Suggest, replace "This API has now modified" to "In the 3.6 release, this API was modified"

**Review by amanda11 — CHANGES_REQUESTED** [review]
Looks good, just a few minor comments.

**Comment by mahesh-orch** [comment]
@amanda11 I have done the changes you suggested. Please review.

**Review by amanda11 — APPROVED** [review]
LGTM

**Comment by amanda11** [comment]
@mahesh-orch I've not merged this yet. Shall we leave open until the rework on the action to make it default to non-deleting files is done, and the documentation updated.
Or do you want to merge this to match what is there now, and open a new one?

**Comment by mahesh-orch** [comment]
@amanda11, @cognifloyd I have updated the docs according to latest changes in action delete API. Please review. Thanks.

File: docs/source/actions.rst
Comment (amanda11) [inline docs/source/actions.rst:78]: Perhaps replace "In the 3.6 release, with kept backward compatibility this API was modified to delete related action files from disk as well."
to 
"In the 3.6 release, with backward compatibility retained this API was modified to optionally delete related action files from disk as well."

File: docs/source/upgrade_notes.rst
Comment (amanda11) [inline docs/source/upgrade_notes.rst:11]: Perhaps rephrase " With kept backward compatibility, action delete API has been modified." to " Retaining backwards compatibility, action delete API has been modified."

**Review by amanda11 — CHANGES_REQUESTED** [review]
Just some minor comments.
However, we could also argue that as this is now backwards compatible it doesn't need to be so detailed in the documents. But I'll leave that to others for their thoughts.

**Comment by mahesh-orch** [comment]
@amanda11 I have made the your suggested changes. Please review. Thanks.

**Review by amanda11 — APPROVED** [review]
LGTM / thanks for updating documentation

**Review by cognifloyd — APPROVED** [review]

**Pull request merged**
